### PR TITLE
Allow the UserPuckCourseView drawn by the given bounds scale.

### LINKED ISSUE #3444: [Bug]: `UserPuckCourseView` is trimmed when using custom frame.
### Mapbox Navigation SDK version

`v2.0.0-rc.4`

### Steps to reproduce

1. Change `NavigationMapView.userLocationStyle` property to:
```swift
let userPuckCourseView = UserPuckCourseView(frame: CGRect(origin: .zero, size: CGSize(width: 50.0, height: 50.0)))
userPuckCourseView.puckColor = .red
userPuckCourseView.fillColor = .green
userPuckCourseView.shadowColor = .blue
navigationViewController.navigationMapView?.userLocationStyle = .courseView(userPuckCourseView)
```
2. Start active guidance navigation.
3. Observe behavior. For example:
Simulator Screen Shot - iPod touch (7th generation) - 2021-10-06 at 11 23 13

### Expected behavior

`UserPuckCourseView` should not be trimmed in case when custom size is used.

### Actual behavior

`UserPuckCourseView` is trimmed.

### Is this a one-time issue or a repeatable issue?

repeatable

### PR DESCRIPTION
**Pull request by ShanMa1991**
### Description
This pr is to fix #3444 to allow the `UserPuckCourseView` scaled by the customized frame.

### Implementation
Because the `UserPuckStyleKitView` is hard coded in drawing, and we have to update the `UserPuckCourseView`  with the corresponding scale during active navigation, it's better for us to use scale ratio to draw the `UserPuckStyleKitView` instead of transforming the view by the bound size.

### Screenshots or Gifs
We change the `NavigationMapView.userLocationStyle` as following and start active navigation: 
```
let userPuckCourseView = UserPuckCourseView(frame: CGRect(origin: .zero, size: CGSize(width: 50.0, height: 50.0)))
userPuckCourseView.puckColor = .red
userPuckCourseView.fillColor = .green
userPuckCourseView.shadowColor = .blue
navigationViewController.navigationMapView?.userLocationStyle = .courseView(userPuckCourseView)
```
scaledPuck

### CODE REVIEW

**Comment by S2Ler** [comment]
@ShanMa1991 I'm wondering, why snapshot images increased in size? Have you checked it?

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@
 * Added the `ReplayLocationManager.replayCompletionHandler` property that allows you to loop location. ([#3528](https://github.com/mapbox/mapbox-navigation-ios/pull/3528), [3550](https://github.com/mapbox/mapbox-navigation-ios/pull/3550))
 * Changed the behavior of `ReplayLocationManager` so that it doesn't loop locations by default. ([#3550](https://github.com/mapbox/mapbox-navigation-ios/pull/3550))
 * Fixed an issue where `ReplayLocationManager` didn't update location timestamps when a new loop started. ([#3550](https://github.com/mapbox/mapbox-navigation-ios/pull/3550))
+* Fixed an issue where `UserPuckCourseView` is trimmed when using custom frame for `NavigationMapView.userLocationStyle`. ([#3601](https://github.com/mapbox/mapbox-navigation-ios/pull/3601))
```
Comment (MaximAlien) [inline CHANGELOG.md:23]:
```suggestion
* Fixed an issue where `UserPuckCourseView` is trimmed when using custom frame for `UserLocationStyle.courseView(_:)`. ([#3601](https://github.com/mapbox/mapbox-navigation-ios/pull/3601))
```

**Comment by MaximAlien** [comment]
> @ShanMa1991 I'm wondering, why snapshot images increased in size? Have you checked it?

Most likely because of frame we're using in tests. Since puck now fills whole frame there are a lot more non-transparent pixels.

**Review by MaximAlien — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ShanMa1991) [reply CHANGELOG.md]: Thanks for the suggestion, I've updated the changeLog.

**Pull request merged**